homeassistant/components/icloud.py

```diff
@@ -0,0 +1,956 @@
+"""
+Platform that supports scanning iCloud.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/icloud/
+"""
+import logging
+from datetime import datetime, timedelta
+from math import floor
+import random
+
+import voluptuous as vol
+
+from pytz import timezone
+import pytz
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.entity import (Entity, generate_entity_id)
+from homeassistant.components.device_tracker import see
+from homeassistant.helpers.event import (track_state_change,
+                                         track_utc_time_change)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+import homeassistant.util.dt as dt_util
+from homeassistant.util.location import distance
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyicloud==0.9.1']
+
+DEPENDENCIES = ['zone', 'device_tracker']
+
+CONF_EVENTS = 'events'
+DEFAULT_EVENTS = False
+CONF_IGNORED_DEVICES = 'ignored_devices'
+CONF_GMTT = 'google_maps_travel_time'
+CONF_COOKIEDIRECTORY = 'cookiedirectory'
+
+# entity attributes
+ATTR_ACCOUNTNAME = 'account_name'
+ATTR_INTERVAL = 'interval'
+ATTR_DEVICENAME = 'device_name'
+ATTR_BATTERY = 'battery'
+ATTR_DISTANCE = 'distance'
+ATTR_STARTTIME = 'start_time'
+ATTR_ENDTIME = 'end_time'
+ATTR_DURATION = 'duration'
+ATTR_REMAINING = 'remaining_time'
+ATTR_DEVICESTATUS = 'device_status'
+ATTR_LOWPOWERMODE = 'low_power_mode'
+ATTR_BATTERYSTATUS = 'battery_status'
+ATTR_LOCATION = 'location'
+ATTR_FRIENDLY_NAME = 'friendly_name'
```

Comment:
You should never have to use this. Just have your entities have a name and `update_ha_state` will take care of this.